atlasdb-workload-server-api/src/main/java/com/palantir/atlasdb/workload/invariant/InvariantReporter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.invariant;
+
+import com.palantir.atlasdb.workload.workflow.WorkflowHistory;
+import java.util.function.Consumer;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface InvariantReporter<Violations> {
```

Comment:
nit: `ViolationsT` for style (usually it's either one letter, or `DescriptiveT`) - I tend to read `Violations` as almost like a class itself!